### PR TITLE
Add root cmake

### PR DESCRIPTION
**Pull request by nkh-lab**
Signed-off-by: Mykola Khyliuk <nkh@ua.fm>

### CODE REVIEW

File: projects/android-studio/libasynccpp/build.gradle [hunk 1]
```diff
@@ -44,7 +44,7 @@ android {
 dependencies {
 	implementation fileTree(dir: 'libs', include: ['*.jar'])
 
-	api project(':libdatacpp')
+	//api project(':libdatacpp')
```
Comment (lufinkey) [inline projects/android-studio/libasynccpp/build.gradle:47]: This is required. I need to include java data

File: projects/android-studio/app/build.gradle [hunk 2]
```diff
@@ -23,15 +23,15 @@ android {
...
+            path "../CMakeLists.txt"
             version "3.10.2"
         }
     }
 }
 
 dependencies {
-    api project(':libdatacpp')
-    api project(':libasynccpp')
+    //api project(':libdatacpp')
```
Comment (lufinkey) [inline projects/android-studio/app/build.gradle:33]: This is also required

**Comment by lufinkey** [comment]
This also seems to still give me the error `cannot find -lDataCpp`

File: projects/android-studio/app/build.gradle [hunk 3]
```diff
@@ -23,15 +23,15 @@ android {
...
+    //api project(':libasynccpp')
```
Comment (lufinkey) [inline projects/android-studio/app/build.gradle:34]: This is required for hooking into the Looper class

**Comment by lufinkey** [comment]
I'm pretty sure it has to do with the fact that they aren't in the same tree structure. I specify the binary directory, but I don't think that's enough somehow

**Comment by lufinkey** [comment]
side note, thank you for taking the time to make this PR. If you're able to get it resolved I can venmo or cashapp you some money

**Comment by nkh-lab** [comment]
I will try to help you.

For now I see next error during compiling the project:
`
In file included from /home/nikolay/Projects/My/async-cpp/src/fgl/async/DispatchWorkItem.cpp:9:
In file included from /home/nikolay/Projects/My/async-cpp/src/fgl/async/DispatchWorkItem.hpp:11:
/home/nikolay/Projects/My/async-cpp/src/fgl/async/Common.hpp:20:11: fatal error: 'fgl/data.hpp' file not found
`
So, it seems that such stuff is missed.

**Comment by lufinkey** [comment]
You need both data-cpp and async-cpp to be cloned into the same folder

**Comment by nkh-lab** [comment]
I know nothing about it, what I'm expecting when I clone some unknown for me repo which has external dependencies that it should at least describe it from where I should fetch it or better to resolve it via using git submodule feature.

**Comment by lufinkey** [comment]
Well that's not how I structured it sorry. I didn't know someone was going to make a PR on the library itself. That's why I opened a stack overflow post.

It's just a library that
is supposed to be pulled in next to its dependencies:

**Comment by lufinkey** [comment]
Libraries Folder
- data-cpp
- async-cpp

Can you clone it like this and please try ^?

**Comment by nkh-lab** [comment]
> data-cpp
> async-cpp
> Can you clone it like this and please try ^?

Sorry, I don't understand how can I clone them? are they separate projects?

Give me the link where can I find 'fgl/data.hpp.

**Comment by lufinkey** [comment]
It's just in the same place you found this one
https://github.com/lufinkey/data-cpp

**Comment by nkh-lab** [comment]
I don't see there fgl/data.hpp file.

**Comment by lufinkey** [comment]
You don't need to find the specific header. I promise it's there. Just clone them next to each other and it'll work.

**Comment by nkh-lab** [comment]
As I mentioned build highlights it like error:  https://github.com/lufinkey/async-cpp/pull/1#issuecomment-865779857

https://github.com/lufinkey/async-cpp/blob/43d51be4197aa2192fb11b4d534fe8bd5487ce61/src/fgl/async/Common.hpp#L20

**Comment by lufinkey** [comment]
Did you clone async-cpp and data-cpp into folders next to each other? If
yes, then it likely is an issue with where the CMakeLists.txt files got
moved on your branch. Try it on master. If no, you need to clone those next
to each other.

The header is in the `src/fgl` folder of data-cpp and that folder is added as an include path on async-cpp. That's why they need to be cloned into the same folder

**Comment by lufinkey** [comment]
The structure is just

- some project folder:
  - data-cpp: (https://github.com/lufinkey/data-cpp)
  - async-cpp: (https://github.com/lufinkey/async-cpp) 

They're meant to be imported into another project, but I have a test app inside each of them so that I can test out library functionality

**Comment by lufinkey** [comment]
Closing this as it's now fixed. Android requires you to specify the entire build path of the library (`${DATACPP_ROOT}/projects/android-studio/libdatacpp/build/intermediates/cmake/${CMAKE_BUILD_TYPE}/obj/${ANDROID_ABI}/libDataCpp`)

**Pull request closed without merging**